templates/cloud_init/cloud_init.client.yml

```diff
@@ -0,0 +1,374 @@
+
+      echo "terraform init"
+      terraform init
+      [ $? -ne 0 ] && error "terraform init."
+
+      echo "terraform apply"
+      terraform apply -auto-approve |& tee /tmp/scaledown.txt
+      [ $? -ne 0 ] && error "terraform apply."
+
+      echo "sleep 1 second"
+      sleep 1
+
+  - path: /opt/destroy.sh
+    permissions: '0755'
+    content: |
+      #!/bin/bash
+
+      function error() {
+        >&2 echo "echo >< ERROR: $* See log for details."
+        exit 1
+      }
+
+      export ARM_SUBSCRIPTION_ID='<sub_id>'
+      export ARM_CLIENT_ID='<arm_client_id>'
+      export ARM_CLIENT_SECRET='<arm_client_scrt>'
+      export ARM_TENANT_ID='<arm_tenant_id>'
+      export TF_VFXT_CONFIG='<terraform_file_config>'
+
+      echo "to file path"
+      cd /tf/src/terraform/examples/vfxt/${TF_VFXT_CONFIG}
+      [ $? -ne 0 ] && error "filepath."
+
+      echo "terraform init"
+      terraform init
+      [ $? -ne 0 ] && error "terraform init."
+
+      echo "terraform destroy"
+      terraform destroy -auto-approve |& tee /tmp/destroy.txt
+      [ $? -ne 0 ] && error "terraform destroy."
+
+      echo "sleep 1 second"
+      sleep 1
+
+  - path: /opt/vdbench_destroy.sh
+    permissions: '0755'
+    content: |
+      #!/bin/bash
+
+      function error() {
+        >&2 echo "echo >< ERROR: $* See log for details."
+        exit 1
+      }
+
+      export ARM_SUBSCRIPTION_ID='<sub_id>'
+      export ARM_CLIENT_ID='<arm_client_id>'
+      export ARM_CLIENT_SECRET='<arm_client_scrt>'
+      export ARM_TENANT_ID='<arm_tenant_id>'
+      export REGION='<region>'
+      export RESOURCE_GROUP_NAME='<resource_group_name>'
+      export TF_VFXT_CONFIG='<terraform_file_config>'
+
+      echo "to file path"
+      cd /tf/src/terraform/examples/vfxt/${TF_VFXT_CONFIG}
+      [ $? -ne 0 ] && error "filepath."
+
+      echo "terraform outputs"
+      CONTROLLER_ADDRESS=$(terraform output -json | jq -r ".controller_address.value")
+      USERNAME=$(terraform output -json | jq -r ".controller_username.value")
+      VSERVER_IP_ADDRESSES=$(terraform output -json | jq -c ".mount_addresses.value")
+      NFS_EXPORT_PATH=$(terraform output -json | jq -r ".mount_namespace_path.value")
+      [ $? -ne 0 ] && error "outputs."
+
+      echo "sskey"
+      sshkey="<ssh_key> azureuser@linuxvm"
+      [ $? -ne 0 ] && error "sshkey."
+
+      cd ../vdbench
+
+      echo "terraform init"
+      terraform init
+      [ $? -ne 0 ] && error "terraform init."
+
+      echo "terraform destroy"
+      terraform destroy -auto-approve -var="location=${REGION}" \
+        -var="vmss_resource_group_name=${RESOURCE_GROUP_NAME}-vmss" \
+        -var="controller_address=${CONTROLLER_ADDRESS}" \
+        -var="controller_username=${USERNAME}" \
+        -var="ssh_key_data=${sshkey}" \
+        -var="vserver_ip_addresses=${VSERVER_IP_ADDRESSES}" \
+        -var="nfs_export_path=${NFS_EXPORT_PATH}" \
+        -var="vdbench_url=<vdbench_download_url>" \
```

Comment:
Unsure what you mean, atleast from what I know, I don't think you can make variables needed only at apply or destroy.

Reply:
Not necessarily for this PR, but this is one option to consider: [Assign variables with tfvars file](https://learn.hashicorp.com/tutorials/terraform/variables#assign-values-with-a-terraform-tfvars-file)

This would simplify the code a bit where you could generate the tfvars file once and use it in multiple places. But not much savings, so probably not something for now. Maybe a follow-up PBI?